screensaver/screensaver.py

```diff
@@ -74,86 +93,269 @@ def get_last_level_directories(photo_dirs_list) -> list:
             if line in line2:
                 counter += 1
         if counter == 1:
-            photo_dirs_list_final.append(line.rstrip())
+            final_dir = os.path.join(line.rstrip(), '')
+            photo_dirs_list_final.append(final_dir)
     return photo_dirs_list_final
 
 
-def delete_old_screensaver_photos() -> None:
+def rsync_directory(source, destination) -> None:
+
+    # run rsync
+    proc = subprocess.run(["rsync",
+                           # verbose
+                           "-v",
+                           # recursive
+                           "-r",
+                           # protect arguments --protect-args equivalent
+                           "-s",
+                           # skip based on checksum, not mod-time & size
+                           "-c",
+                           # delete extraneous files from dest dirs
+                           "--delete",
+                           # use special port
+                           "--rsh", "ssh -p" + RSYNC_PORT,
+                           # include files:
+                           "--include", regex_files,
+                           # exclude files:
+                           "--exclude", "*",
+                           # from:
+                           source,
+                           # to:
+                           destination
+                           ],
+                          stdout=subprocess.PIPE,  # required to return names of photos transferred
+                          universal_newlines=True  # required to return string, not bytes-like obj
+                          # stderr=subprocess.STDOUT
+                          )
+    pass
+
+
+def copy_directory_locally(source, destination):
```

Comment:
Please use [function annotation](https://www.python.org/dev/peps/pep-3107/) to specify the return type of the function, and also add a doc string to summarise what it does.